TUDelft-CNS-ATM/bluesky
# TerrainTerrain elevation and associated functionalities added with TERRAIN plugin v1.0

Dear BlueSky Community,

I have implemented the initial version of the plugin `TERRAIN`.

To test `TERRAIN` on your end,

1. configure the `terrain_and_skyline` section in `settings.cfg`. You may want to request an [Open Topography](https://opentopography.org/) key for free, and use it to replace the default key.

2. Run `plugin terrain` in BlueSky console. Stand by for the plugin to download necessary dataset and initialize its functions.

3.  Create an aircraft within the set bounding box. E.g., a 747 over Vienna,
`cre A b744 48.2, 16.3 0m 10000 200`

4. `pan A`.

5. Run `AGL A` for the updated above-ground level of `A`. You may hear a sound that says the AGL value in feet 🙄🙄🙄

6. Mute that sound  in `settings.cfg` with `agl_mute = True` 🤣🤣🤣

7. Run `groundmsl A` or `groundmsl 48.2, 16.3` to check the ground elevation below `A` or at `(lat, lon)` = `(48.2, 16.3)`.

8. Run `SHOWELEMAP` to print a drawing of the terrain elevation map in .pdf and .png formats.

Please kind let me know if you have spotted any issue for me to fix, especially if I have broken any convention of BlueSky that I am not aware of.

As always, suggestions for me to improve and expand the plugin `TERRAIN` are much welcomed.

Cheers,

Yaofu

P.s. we can totally tell landing aircraft to say "50, 40, 30, 20, 10, retard, retard, retard..." if you want!
